File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Foundation/NoDuplicatesQueueTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4objects.Db4o.Foundation;

namespace Db4objects.Db4o.Tests.Common.Foundation
{
	public class NoDuplicatesQueueTestCase : ITestLifeCycle
	{
		private IQueue4 _queue;

		public virtual void Test()
		{
			_queue.Add("A");
			_queue.Add("B");
			_queue.Add("B");
			_queue.Add("A");
			Assert.AreEqual("A", _queue.Next());
			Assert.AreEqual("B", _queue.Next());
			Assert.IsFalse(_queue.HasNext());
		}

		/// <exception cref="System.Exception"></exception>
		public virtual void SetUp()
		{
			_queue = new NoDuplicatesQueue(new NonblockingQueue());
		}

		/// <exception cref="System.Exception"></exception>
		public virtual void TearDown()
		{
			_queue = null;
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/foundation/NoDuplicatesQueueTestCase.java
/* Copyright (C) 2008   Versant Inc.   http://www.db4o.com */

package com.db4o.db4ounit.common.foundation;

import com.db4o.foundation.*;

import db4ounit.*;

public class NoDuplicatesQueueTestCase implements TestLifeCycle {

	private Queue4 _queue;

	public void test() {
		_queue.add("A");
		_queue.add("B");
		_queue.add("B");
		_queue.add("A");
		Assert.areEqual("A", _queue.next());
		Assert.areEqual("B", _queue.next());
		Assert.isFalse(_queue.hasNext());
	}
	
	public void setUp() throws Exception {
		_queue = new NoDuplicatesQueue(new NonblockingQueue());
	}

	public void tearDown() throws Exception {
		_queue = null;
	}
}
